apps/codeimage/src/core/modules/pagination/createPagedData.ts

```diff
@@ -0,0 +1,26 @@
+import {Accessor, createSignal, Setter, createEffect} from 'solid-js';
+
+export const createPagedData = <T>(
+  data: Accessor<T[] | undefined>,
+): [
+  Accessor<T[]>,
+  {page: Accessor<number>; setPage: Setter<number>; pageSize: number},
+] => {
+  const [page, setPage] = createSignal(1),
+    pageSize = 9;
```

Comment:
The page size must be passed as option

export const createPagedData = <T>(
  data: Accessor<T[] | undefined>,
  options: {
    pageSize: number
  }
): [
  Accessor<T[]>,
  {page: Accessor<number>; setPage: Setter<number>; pageSize: number},
]